### PR TITLE
Warnings fixes in DAL, XUnitTest and XIntegrationTest layers

### PR DESCRIPTION
**Pull request by mythter**
* Ticket [#1651](https://github.com/ita-social-projects/StreetCode/issues/1651)

## Code reviewers

- [ ] @sashapanasiuk5 
- [x] @Mikhail-Kolpakov 
- [ ] @ita-social-projects/734-08-streetcode 

## Summary of issue

- Check and resolve any warnings present in the application codebase (DAL, XUnitTest and XIntegrationTest layers). This includes identifying warnings reported by the compiler, static analysis tools, or any other code quality tools integrated into the development environment.

## Summary of change

- Fixed warnings in the specified projects
- [DAL MimeKit nuget package updated to 4.7.1](https://github.com/ita-social-projects/StreetCode/commit/ac8fdb3a31301a60baacec59c2776d7f07f6eb55)
- Newss changed to News where needed
- Commented tests were uncommented and made functional (or partially)
- [BLL all null checks in GetAll handlers changed to .Any() method check](https://github.com/ita-social-projects/StreetCode/commit/e5db82a3c1fe25396f11c62690e56d088e40f9fe)
- Fixed tests that started fail after BLL fixes

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ArtControllerTests.cs [hunk 1]
```diff
@@ -64,19 +67,20 @@ public async Task GetById_ReturnSuccessStatusCode()
...
+        }
+
+        // [Fact(Skip = "There is no Art that has StreetcodeArts, so it will fail without them.")]
         [Fact]
         public async Task GetArtsByStreetcodeId_ReturnSuccessStatusCode()
         {
-            ArtExtracter.AddStreetcodeArt(this._testStreetcodeContent.Id, this._testArt.Id);
-            int streetcodeId = this._testStreetcodeContent.Id;
-            var response = await this.client.GetArtsByStreetcodeId(streetcodeId);
+            ArtExtracter.AddStreetcodeArtWithStreetcodeArtSlide(this.testStreetcodeContent.Id, this.testArt.Id, this.testStreetcodeArtSlide.Id);
```
Comment (Mikhail-Kolpakov) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ArtControllerTests.cs:81]: Same typo I mentioned above.

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ImageControllerTests.cs [hunk 2]
```diff
@@ -30,14 +30,14 @@ public ImageControllerTests(CustomWebApplicationFactory<Program> factory)
 
         public override void Dispose()
         {
-            StreetcodeContentExtracter.Remove(this._testStreetcodeContent);
-            ImageExtracter.Remove(this._testImage);
+            StreetcodeContentExtracter.Remove(this.testStreetcodeContent);
+            ImageExtracter.Remove(this.testImage);
```
Comment (Mikhail-Kolpakov) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ImageControllerTests.cs:34]: Same typo I mentioned above. Check this kind of typo in whole your code.

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Base/BaseClient.cs [hunk 3]
```diff
@@ -5,23 +5,23 @@ namespace Streetcode.XIntegrationTest.ControllerTests.Utils.Client.Base
...
 
         public BaseClient(HttpClient client, string secondPartUrl = "")
         {
-            _client = new RestClient(client) { AcceptedContentTypes = ContentType.JsonAccept };
-            _secondPartUrl = secondPartUrl;
+            this.client = new RestClient(client) { AcceptedContentTypes = ContentType.JsonAccept };
+            this.SecondPartUrl = secondPartUrl;
         }
 
+        protected string SecondPartUrl { get; }
```
Comment (Mikhail-Kolpakov) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Base/BaseClient.cs:16]: Better to write this property before constructor (under `private readonly RestClient client;` prop).

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Extracter/StreetcodeExtracter/StreetcodeUpdateDTOExtracter.cs [hunk 4]
```diff
@@ -17,13 +17,17 @@
 
 namespace Streetcode.XIntegrationTest.ControllerTests.Utils.Extracter.StreetcodeExtracter
 {
-    public class StreetcodeUpdateDTOExtracter
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
```
Comment (Mikhail-Kolpakov) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Extracter/StreetcodeExtracter/StreetcodeUpdateDTOExtracter.cs:20]: Same issue I mentioned above.

File: Streetcode/Streetcode.XIntegrationTest/ServiceTests/BlobServiceTests/Utils/BlobStorageFixture.cs [hunk 5]
```diff
@@ -1,55 +1,56 @@
...
-            Directory.CreateDirectory(blobPath);
+            this.BlobService = new BlobService(this.environmentVariables); // add repo
+            Directory.CreateDirectory(this.BlobPath);
         }
 
+        public BlobService BlobService { get; }
+
+        public string BlobPath { get; private set; }
+
+        public StreetcodeDbContext TestDbContext { get; private set; } = null!;
```
Comment (Mikhail-Kolpakov) [inline Streetcode/Streetcode.XIntegrationTest/ServiceTests/BlobServiceTests/Utils/BlobStorageFixture.cs:33]: Better to write this properties before constructor (check in other places in your new code).

**Review by Mikhail-Kolpakov — APPROVED** [review]

File: Streetcode/Streetcode.DAL/Persistence/StreetcodeDbContext.cs [hunk 6]
```diff
@@ -75,61 +75,61 @@ public StreetcodeDbContext(DbContextOptions<StreetcodeDbContext> options)
     public DbSet<TeamMemberPositions> TeamMemberPosition { get; set; }
     public DbSet<Job> Job { get; set; }
 
-    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    protected override void OnModelCreating(ModelBuilder builder)
```
[the remark this answers is not in the document]
Comment (mythter) [reply Streetcode/Streetcode.DAL/Persistence/StreetcodeDbContext.cs:78]: There was a warning that the parameter name should be the same as in the base class method

File: Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs [hunk 7]
```diff
@@ -17,5 +15,7 @@ public IntegrationTestBase()
 
             this.Configuration = configBuilder.Build();
         }
+
+        protected IConfigurationRoot Configuration { get; }
```
[the remark this answers is not in the document]
Comment (mythter) [reply Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs:19]: Yeah, I know. I just fixed the warning [SA1201](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation%2FSA1201.md)

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs [hunk 8]
```diff
@@ -1,39 +1,44 @@
...
-        private StreetcodeContent _testStreetcodeContent;
+        private readonly StreetcodeContent testStreetcodeContent;
+        private readonly StreetcodeCoordinate testStreetcodeCoordinate;
 
         public CoordinateControllerTests(CustomWebApplicationFactory<Program> factory)
             : base(factory, "/api/Coordinate")
         {
             int uniqueId = UniqueNumberGenerator.GenerateInt();
-            this._testStreetcodeContent = StreetcodeContentExtracter
+            this.testStreetcodeContent = StreetcodeContentExtracter
```
[the remark this answers is not in the document]
Comment (mythter) [reply Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs:23]: This spelling uses across the project. So I decided not to touch it

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Base/BaseClient.cs (hunk 3 above)
Comment (mythter) [reply Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/Client/Base/BaseClient.cs:16]: [SA1201](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation%2FSA1201.md)

File: Streetcode/Streetcode.XIntegrationTest/Constants/ControllerTests/AuthConstants.cs [hunk 9]
```diff
@@ -4,6 +4,10 @@
 
 namespace Streetcode.XIntegrationTest.Constants.ControllerTests
 {
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
```
[the remark this answers is not in the document]
Comment (mythter) [reply Streetcode/Streetcode.XIntegrationTest/Constants/ControllerTests/AuthConstants.cs]: Fixed it

File: Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (Mikhail-Kolpakov) [reply Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs:19]: Wouldn't be better just to suppress this warning globally like you tit in `.csproj`? Or at least use `[SuppressMessage...]` attribute in some places.

image

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (Mikhail-Kolpakov) [reply Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs:23]: Let it be.

File: Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs (hunk 7 above)
[the remark this answers is not in the document]
Comment (mythter) [reply Streetcode/Streetcode.XIntegrationTest/Base/IntegrationTestBase.cs:19]: Maybe, we can discuss this on the meeting

**Review by DDeviak — APPROVED** [review]
LGTM

**Review by ulyasaur — APPROVED** [review]

**Pull request merged**